docs/using-hive.md

```diff
@@ -2,72 +2,231 @@
+
+```yaml
+apiVersion: v1
+data:
+  aws_access_key_id: REDACTED
+  aws_secret_access_key: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-aws-creds
+  namespace: hive
+type: Opaque
+```
+
+#### Azure
+
+Create a `secret` containing your Azure service principal:
+
+```yaml
+apiVersion: v1
+data:
+  osServicePrincipal.json: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-azure-creds
+  namespace: hive
+type: Opaque
+```
+
+#### GCP
+
+Create a `secret` containing your GCP service account key:
+
+```yaml
+apiVersion: v1
+data:
+  osServiceAccount.json: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-gcp-creds
+  namespace: hive
+type: Opaque
+```
+
+### SSH Key Pair
+
+(Optional) Hive uses the provided ssh key pair to ssh into the machines in the remote cluster. Hive connects via ssh to gather logs in the event of an installation failure. The ssh key pair is optional, but neither the user nor Hive will be able to ssh into the machines if it is not supplied.
+
+Create a Kubernetes secret containing a ssh key pair (typically generated with `ssh-keygen`)
+
+```yaml
+apiVersion: v1
+data:
+  ssh-privatekey: REDACTED
+  ssh-publickey: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-ssh-key
+  namespace: hive
+type: Opaque
+```
+
+### InstallConfig
+
+The OpenShift installer `InstallConfig` must be base64 encoded into a `secret` and referenced in the `ClusterDeployment`.
+
+```yaml
+apiVersion: v1
+baseDomain: hive.example.com
+compute:
+- name: worker
+  platform:
+    aws:
+      rootVolume:
+        iops: 100
+        size: 22
+        type: gp2
+      type: m4.xlarge
+    gcp:
+      type: n1-standard-4
+    azure:
+      osDisk:
+        diskSizeGB: 128
+      type: Standard_D2s_v3
+  replicas: 3
+controlPlane:
+  name: master
+  platform:
+    aws:
+      rootVolume:
+        iops: 100
+        size: 22
+        type: gp2
+      type: m4.xlarge
+    gcp:
+      type: n1-standard-4
+    azure:
+      osDisk:
+        diskSizeGB: 128
+      type: Standard_D2s_v3
+replicas: 3
+metadata:
+  creationTimestamp: null
+  name: mycluster
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineCIDR: 10.0.0.0/16
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  aws:
+    region: us-east-1
+  gcp:
+    projectID: myproject
+    region: us-east1
+  azure:
+    baseDomainResourceGroupName: my-bdrgn
+    region: centralus
+pullSecret: REDACTED
+sshKey: REDACTED
 ```
 
-Alternatively you can specify release image overrides directly on your ClusterDeployment when you create one.
+```bash
+base64 install-config.yaml
+```
 
-Cluster provisioning begins when a caller creates a ClusterDeployment CR, which is the core Hive resource to control the lifecycle of a cluster.
+```yaml
+apiVersion: v1
+data:
+  install-config.yaml: BASE64 ENCODED INSTALLCONFIG
+kind: Secret
+metadata:
+  name: mycluster-install-config
+  namespace: hive
+type: Opaque
+```
```

Comment:
Couple ideas for more elegant way to do this:

1. Show a secret with stringData embedding the InstallConfig all in one yaml document.
2. `kubectl create secret generic mycluster-install-config --from-file=./install-config.yaml` (untested but something like that should work)

Reply:
done, option 2 :)